airflow/config_templates/config.yml

```
@@ -944,6 +944,14 @@ metrics:
   description: |
     StatsD (https://github.com/etsy/statsd) integration settings.
   options:
+    metrics_use_pattern_match:
+      description: |
+        If true, metrics_allow_list and metrics_block_list will use regex pattern matching in the metric
+        name and not just start of the name.
```

A suggestion to make this even more concrete and spelled out. Seems obvious to us working on it, but it's a slightly tricky one if you aren't familiar.
```suggestion
        If true, metrics_allow_list and metrics_block_list will use regex pattern matching anywhere within the metric
        name instead of only prefix matching at the start of the name.
```